### ISSUE TITLE
Dependency on conda-standalone for osx-arm64 breaks install

### ISSUE THREAD

**Issue by FaustinCarter** [conda/constructor#742]
### Checklist

- [X] I added a descriptive title
- [X] I searched open reports and couldn't find a duplicate

### What happened?

TL;DR: Can we remove the hard dependency on conda-standalone for Unix-like systems (linux, osx, etc) since there are other options available (like micromamba).

Attempting to install constructor on an arm64 mac results in a cryptic error message:

```
LibMambaUnsatisfiableError: Encountered problems while solving:
  - nothing provides __win needed by constructor-3.3.1-pyh95a074a_2
```

This is because for some reason conda is attempting to install an older Windows build of constructor instead of the latest Unix build. When forcing conda to install the latest Unix build, the real issue becomes clear:

```
LibMambaUnsatisfiableError: Encountered problems while solving:
  - nothing provides conda-standalone needed by constructor-3.5.0-pyhe4f9e05_0
```

The root cause is that there _is_ no build of conda-standalone for arm64 Mac because the following PR on the feedstock has been hung up for some time https://github.com/conda-forge/conda-standalone-feedstock/pull/54.

However, this isn't an actual problem for _using_ constructor on arm64 Macs, because one can use micromamba in place of conda-standalone (and indeed that's just what conda-forge does). I rebuilt constructor locally to remove the dependency, and then used the resulting package (and micromamba) to create an installer for both arm64 and intel macs on a M1 macbook and the installers worked just fine.

So I would propose adding `# [win]` to the end of the conda-standalone dependency line to make it only enforced on a Windows install. I would further propose adding a check in the code that emits a warning if constructor is imported and conda-standalone is not installed.

**Comment by jaimergp** [comment]
I don't think we can / should drop the dependency because many workflows depend on that assumption right now and would cause a lot of churn (even if the fix is super simple). We would need to start adding a warning in the next release and so on.

One less painful solution is to create separate noarch build for constructor osx-arm64, but I don't know how well spread archspec is yet.

A hackier one is to provide an _empty_ conda-standalone for osx-arm64 while we figure https://github.com/conda-forge/conda-standalone-feedstock/pull/54.

We can also build conda-standalone manually for osx-arm64.

Finally, we could copy it from [defaults](https://anaconda.org/main/conda-standalone) now that the feedstocks have been standardized to an extent.

Til any of that happens, the solution is to create a rosetta env for constructor

```
$ conda create -n constructor-x64 --platform=osx-64 constructor
```

**Comment by jaimergp** [comment]
> We can also build conda-standalone manually for osx-arm64.

See https://github.com/conda-forge/conda-standalone-feedstock/pull/69#issuecomment-1893415864

**Comment by jaimergp** [comment]
Closing as per https://github.com/conda-forge/conda-standalone-feedstock/pull/71#issuecomment-1894168073.
